src/compile/data/data.ts

```diff
@@ -48,92 +50,211 @@ export interface DataComponent {
   timeUnitDomain: StringSet;
 
-  /** Array of summary component object for producing summary (aggregate) data source */
-  summary: SummaryComponent[];
+  /** Array of summary component object for producing aggregates */
+  aggregate: AggregateComponent;
 }
 
 /**
- * Composable component for a model's summary data
+ * Component for aggregations.
  */
-export interface SummaryComponent {
-  /** Name of the summary data source */
-  name: string;
-
+export interface AggregateComponent {
   /** String set for all dimension fields  */
   dimensions: StringSet;
 
   /** dictionary mapping field name to string set of aggregate ops */
   measures: Dict<StringSet>;
+
+  /**
+   * Whether we need the unaggrgeated data source.
+   * Should be set when we parse the scales.
+   */
+  assembleRaw?: boolean;
+}
+
+export interface SourceComponent {
+  source?: string;
+  values?: any;
+  format?: any;
+  url?: any;
 }
 
 // TODO: split this file into multiple files and remove this linter flag
 /* tslint:disable:no-use-before-declare */
 
-export function parseUnitData(model: UnitModel): DataComponent {
+/**
+ * Parses a model with data.
+ */
+function parseData(model: Model): DataComponent {
   return {
+    source: source.parseUnit(model),
     formatParse: formatParse.parseUnit(model),
-    nullFilter: nullFilter.parseUnit(model),
+    calculate: calculate.parseUnit(model),
     filter: filter.parseUnit(model),
-    nonPositiveFilter: nonPositiveFilter.parseUnit(model),
-
-    source: source.parseUnit(model),
     bin: bin.parseUnit(model),
-    calculate: formula.parseUnit(model),
     timeUnit: timeUnit.parseUnit(model),
-    timeUnitDomain: timeUnitDomain.parseUnit(model),
-    summary: summary.parseUnit(model),
+    aggregate: aggregate.parseUnit(model),
+    colorRank: colorRank.parseUnit(model),
     stackScale: stackScale.parseUnit(model),
-    colorRank: colorRank.parseUnit(model)
+    nonPositiveFilter: nonPositiveFilter.parseUnit(model),
+    nullFilter: nullFilter.parseUnit(model),
+    timeUnitDomain: timeUnitDomain.parseUnit(model)
   };
 }
 
+/**
+ * Create components for data source and transforms/ aggregates.
+ */
+export function parseUnitData(model: UnitModel): DataComponent {
+  return parseData(model);
+}
+
+/**
+ * The data component for a facet is the data component of its child.
+ * We only need to add the facet fields as dimensions to the summary component.
+ */
 export function parseFacetData(model: FacetModel): DataComponent {
+  const child = model.child();
+
+  // TODO: enforce this in the schema
+  if (child.data() || !empty(child.transform())) {
+    model.addWarning('Facet child should not define data source or transform.');
+  }
+
+  // rename all references to child data, everything should have been merged up
+  [RAW, SOURCE, STACKED_SCALE].forEach((data) => {
+    model.renameData(child.dataName(data), model.dataName(data));
+  });
+
+  // we should completely move the data component to the facet because
```

Comment:
not sure which part of code this comment try to address (above? below? or is it a TODO?)